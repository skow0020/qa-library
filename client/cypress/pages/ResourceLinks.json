{
    "addResourceLink": "#add-resourceLink",
    "firstResourceLinkCard": "#resource-card-0",
    "resourceLinkCards": "[id*=resource-card]",
    "category": "#category",
    "language": "#language",
    "cardPosts": ".card-post"
}