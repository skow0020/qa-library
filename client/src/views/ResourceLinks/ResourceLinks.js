/* eslint jsx-a11y/anchor-is-valid: 0 */

import {
  Badge,
  Button,
  Card,
  CardBody,
  Col,
  Container,
  Form,
  FormGroup,
  Row
} from "shards-react";

import CategoriesSelection from "../../components/common/CategoriesSelection";
import LanguagesSelection from "../../components/common/LanguagesSelection";
import LoadError from "../../components/common/LoadError";
import Loading from "../../components/common/Loading";
import PageTitle from "../../components/common/PageTitle";
import React from "react";
import { getCategoryTheme } from "../../utils/util";

class ResourceLinks extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      resourceLinks: [],
      isLoading: false,
      filter: null
    };

    this.getResourceLinks = this.getResourceLinks.bind(this);
    this.handleFilterChange = this.handleFilterChange.bind(this);
    this.handleLanguageChange = this.handleLanguageChange.bind(this);
  }

  componentDidMount() {
    this.getResourceLinks();
  }

  getResourceLinks() {
    const categoryFilter = this.state.category ? `category=${this.state.category}` : '';
    const languageFilter = this.state.language ? `language=${this.state.language}` : '';
    const filter = `${categoryFilter}&${languageFilter}`;

    this.setState({ isLoading: true });
    fetch(`/api/resourceLinks?${filter}`)
      .then(response => response.json())
      .then(
        data => this.setState({ resourceLinks: data.data, isLoading: false }),
        error => this.setState({ error, isLoading: false })
      );
  }

  nextPath(path) {
    this.props.history.push(path);
  }

  handleFilterChange(e) {
    this.setState({
      category: e.target.value
    }, () => this.getResourceLinks());
  }

  handleLanguageChange(e) {
    this.setState({
      language: e.target.value
    }, () => this.getResourceLinks());
  }

  render() {
    const { resourceLinks, isLoading, error, category, language } = this.state;

    if (isLoading) return <Loading />;

    if (error) return <LoadError error="Resource Links failed to load" />;

    return (
      <Container fluid className="main-content-container px-3">
        <Row noGutters className="form-inline py-2">
          <PageTitle sm="8" title="Resource Links" className="text-sm-left" />
          <Button id="add-resourceLink" type="button" className="btn btn-success btn-lg" onClick={() => this.nextPath('/add-resourceLink')}>
            Add Resource Link
          </Button>
        </Row>
        <Row noGutters className="form-inline py-2">
          <Form id='filtering-form'>
            <FormGroup >
              <CategoriesSelection value={category} onChange={this.handleFilterChange} />
              <LanguagesSelection value={language} onChange={this.handleLanguageChange} />
            </FormGroup>
          </Form>
        </Row>
        <Row>
          {resourceLinks.map((post, idx) => (
            <Col lg="4" md="6" sm="12" className="mb-4" key={idx}>
              <Card small id={`resource-card-${idx}`} className="card-post card-post--1">
                <a href={post.url}>
                  <div
                    className="card-post__image"
                    style={{ backgroundImage: `url('${post.backgroundImage}')` }}
                  />
                </a>
                <Badge pill className={`card-post__category bg-${getCategoryTheme(post.category)}`}>
                  {post.category}
                </Badge>
                <CardBody>
                  <h5 className="card-title">
                    <a href={post.url} className="text-fiord-blue">
                      {post.title}
                    </a>
                  </h5>
                  <p className="card-text d-inline-block mb-3">{post.body}</p>
                  <span className="text-muted">{post.date}</span>
                </CardBody>
              </Card>
            </Col>
          ))}
        </Row>
      </Container>
    );
  }
}

export default ResourceLinks;
