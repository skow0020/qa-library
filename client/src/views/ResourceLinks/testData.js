export const resourceLinks = {
  data: [
    {
      '_id': '5e220997d7ca7340bcfc9ded',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Corporis laborum assumenda non qui. Soluta facilis dolore doloremque enim facere a accusantium.',
      'res_id': 1000
    },
    {
      '_id': '5e220997d7ca7340bcfc9dee',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Voluptatem qui dolore veritatis ullam tempore eos. Deserunt autem nulla praesentium magnam dolorem.',
      'res_id': 1001
    },
    {
      '_id': '5e220997d7ca7340bcfc9def',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Illo qui consequatur eum. Tenetur sed vel est quod.',
      'res_id': 1002
    }
  ]
};