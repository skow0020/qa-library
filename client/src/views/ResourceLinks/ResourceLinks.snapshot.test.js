import React from 'react';
import ResourceLinks from './ResourceLinks';
import renderer from 'react-test-renderer';

describe('ResourceLinks Snapshot Tests', () => {
  test('ResourceLink snapshot', () => {
    const component = renderer.create(
      <ResourceLinks></ResourceLinks>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});