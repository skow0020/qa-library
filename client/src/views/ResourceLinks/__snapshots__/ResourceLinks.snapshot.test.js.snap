// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceLinks Snapshot Tests ResourceLink snapshot 1`] = `
<div
  className="main-content-container px-4 pb-4 container-fluid"
>
  <div
    className="error"
  >
    <div
      className="error__content"
    >
      <h3>
        Loading...
      </h3>
    </div>
  </div>
</div>
`;
